gnosis/safe-react
# [L2-UX] Refactor network config getters

## What it solves
Resolves #2513

## How this PR fixes it
It adds a refactor on the necessary places to be able to handle the `chainId` internally and hot reloading the necessary components.
To handle the different gateway transactions and avoid mixing them, we had to add `chainId` as parameter for the state, so now the transactions will be stored separately for each instance of a safe also depending on the network.

`gatewayTransactions -> [chainId] -> [safeAddress] -> [history|queue]`

In the safe list when unifiying the localStorage between all networks same pattern could apply. This is still to be done. In this PR we just flush the safe list when changing chain Id and load again the expected ones from the local storage

`safes -> [chainId] _> [safeAddress] -> safeInformation`

Updates onboard to load dynamically. By default Onboard create an unique instance and is stored immutable. We are now wrapping on a function that is able to initialize Onboard again whenever chainId changes. We are using a singleton pattern in order to reuse the Onboard instance while we are working in the same network

## How to test it
1. Open the app
2. By default app will load using the chain selected on build time (old method)
~~3. Make sure that Redux Dev Tools extension is installed in your browser~~
~~4. Use Redux Dev Tools to dispatch the following event. You can dispatch using the icon between the Pin and the Chronometer. Just copy and paste there this content including brackets and click dispatch. You could play around changing the `payload` to the desired chainId (1 - Mainnet, 4 - Rinkeby, 100 - xDai...)~~
~~{
type: 'config/setChainId',
payload: '1'
}~~
3. Use the network switcher button to change network


The NetworkSwitcher button will have the NEW behavior in all environments EXCEPT tagged builds (production deployments) and release builds (for release regression)
